tests/views/test_registry_instrument_view.py

```diff
@@ -0,0 +1,206 @@
+                exercise_id, form_type_does_not_exist
+            )
+
+    def test_get_registry_instruments_returns_200_and_empty_list(self):
+        with patch(
+            "application.views.registry_instrument_view.RegistryInstrument.get_registry_instruments_by_exercise_id"
+        ) as mock_get_registry_instruments_by_exercise_id:
+            mock_get_registry_instruments_by_exercise_id.return_value = []
+            response = self.client.get(
+                f"{api_root}/registry-instrument/exercise-id/{exercise_id}",
+                headers=self.get_auth_headers(),
+            )
+            self.assertStatus(response, 200)
+            self.assertEqual(response.json, [])
+            self.assertEqual(response.headers["Content-Type"], "application/json")
+            mock_get_registry_instruments_by_exercise_id.assert_called_once_with(exercise_id)
+
+    def test_get_registry_instruments_returns_200_and_list_of_objects(self):
+        with patch(
+            "application.views.registry_instrument_view.RegistryInstrument.get_registry_instruments_by_exercise_id"
+        ) as mock_get_registry_instruments_by_exercise_id:
+            with open(Path(__file__).parent.parent / "test_data" / "registry_instruments.json") as f:
+                mock_get_registry_instruments_by_exercise_id.return_value = json.load(f)
+            response = self.client.get(
+                f"{api_root}/registry-instrument/exercise-id/{exercise_id}",
+                headers=self.get_auth_headers(),
+            )
+            self.assertStatus(response, 200)
+            self.assertEqual(len(response.json), 2)  # The 2 registry instrument objects in the test data
+            self.assertEqual(response.headers["Content-Type"], "application/json")
+            mock_get_registry_instruments_by_exercise_id.assert_called_once_with(exercise_id)
+
+    def test_successful_delete_registry_instrument_returns_200(self):
+        with patch(
+            "application.views.registry_instrument_view."
+            "RegistryInstrument.delete_registry_instrument_by_exercise_id_and_formtype"
+        ) as mock_delete_registry_instrument_by_exercise_id_and_formtype:
+            mock_delete_registry_instrument_by_exercise_id_and_formtype.return_value = True
+            response = self.client.delete(
+                f"{api_root}/registry-instrument/exercise-id/{exercise_id}/formtype/{form_type_exists}",
+                headers=self.get_auth_headers(),
+            )
+            self.assertEqual(response.data.decode(), "Successfully deleted registry instrument")
+            self.assertEqual(response.headers["Content-Type"], "text/html; charset=utf-8")
+            mock_delete_registry_instrument_by_exercise_id_and_formtype.assert_called_once_with(
+                exercise_id, form_type_exists
+            )
+
+    def test_unsuccessful_delete_registry_instrument_returns_404(self):
```

Comment:
Not a huge fan of this name, Im assuming it's because the registry_instrument isn't in the database, but the name is a bit verbose. It makes me think any failure is coming back 404. test_unsuccessful_delete_registry_instrument_missing_instrument

Reply:
Yeah, agree its a bit naff... I'll change it to your suggestion.